packages/webdriverio/src/commands/element/getCSSProperty.ts

```diff
@@ -64,50 +66,137 @@ import { parseCSS } from '../../utils/index.js'
+    }
+
+    const cssValues = await Promise.all(
+        properties.map((prop) => this.getElementCSSValue(this.elementId, prop))
+    )
+
+    return mergeEqualSymmetricalValue(cssValues)
+}
+
+async function getPropertyCSSValue(
+    this: WebdriverIO.Element,
+    options: Options,
 ) {
+    const { pseudoElement, cssProperty } = options
+
+    if (pseudoElement) {
+        return await getPseudoElementCSSValue(
+            this,
+            {
+                pseudoElement,
+                cssProperty
+            }
+        )
+    }
+    return await this.getElementCSSValue(this.elementId, cssProperty)
+}
+
+function getShorthandProperties(cssProperty: string) {
     /**
      * Getting the css value of a shorthand property results in different results
      * given that the behavior of `getComputedStyle` is not defined in this case.
      * Therefore if we don't deal with a shorthand property run `getElementCSSValue`
      * otherwise expand it and run the command for each longhand property.
      */
-    if (!cssShorthandProps.isShorthand(cssProperty)) {
-        const cssValue = await this.getElementCSSValue(this.elementId, cssProperty)
-        return parseCSS(cssValue, cssProperty)
-    }
-
-    const properties = cssShorthandProps.expand(cssProperty)
-    let cssValues = await Promise.all(
-        properties.map((prop) => this.getElementCSSValue(this.elementId, prop))
-    )
+    return cssShorthandProps.expand(cssProperty)
+}
 
+function mergeEqualSymmetricalValue(cssValues: string[]) {
     /**
      * merge equal symmetrical values
      * - e.g. `36px 10px 36px 10px` to `36px 10px`
      * - or `0px 0px 0px 0px` to `0px`
-     */
-    while ((cssValues.length % 2) === 0) {
+    */
+    let newCssValues = [...cssValues]
+    while ((newCssValues.length % 2) === 0) {
         const mergedValues = [
-            cssValues.slice(0, cssValues.length / 2).join(' '),
-            cssValues.slice(cssValues.length / 2).join(' ')
+            newCssValues.slice(0, newCssValues.length / 2).join(' '),
+            newCssValues.slice(newCssValues.length / 2).join(' ')
         ]
 
         const hasEqualProperties = mergedValues.every((v) => v === mergedValues[0])
         if (!hasEqualProperties) {
             break
         }
 
-        cssValues = cssValues.slice(0, cssValues.length / 2)
+        newCssValues = newCssValues.slice(0, newCssValues.length / 2)
     }
 
-    return parseCSS(cssValues.join(' '), cssProperty)
+    return newCssValues.join(' ')
 }
+
+async function getPseudoElementCSSValue (
+    elem: WebdriverIO.Element,
+    options: Required<Options>
+): Promise<string> {
+    const browser = getBrowserObject(elem)
+    const { cssProperty, pseudoElement }  = options
+    const cssValue = await browser.execute(
+        (elem: Element, pseudoElement: string, cssProperty: string) => (window.getComputedStyle(elem, pseudoElement) as any)[cssProperty],
+        elem as unknown as Element,
```

Comment:
I don't know why it was ok for typescript earlier when I was passing browser via params, but now i got error that `WebdriverIO.Element` can't be used as `Element`. 
The only solution i found is this ugly casting

Reply:
Yeah, this is because the browser driver transforms the `WebdriverIO.Element` reference into a web `Element` reference and we haven't found a nice way to teach TypeScript that 😉 I think this is fine for now